bagdiscovery/routines.py

```diff
@@ -38,20 +66,13 @@ def __init__(self):
             if not os.access(dir, os.W_OK):
                 raise BagDiscoveryException("Directory not writable", dir)
 
-    def run(self):
-        bag_ids = []
-        for bag in Bag.objects.filter(process_status=Bag.CREATED):
-            self.bag_name = "{}.tar.gz".format(bag.bag_identifier)
-            if os.path.exists(os.path.join(self.src_dir, self.bag_name)):
-                self.unpack_bag()
-                self.save_bag_data(bag)
-                self.move_bag(bag)
-                bag_ids.append(bag.bag_identifier)
-                bag.process_status = bag.DISCOVERED
-                bag.save()
-            else:
-                continue
-        return ("All bags discovered.", bag_ids)
+    def process_bag(self, bag):
+        self.bag_name = "{}.tar.gz".format(bag.bag_identifier)
+        if os.path.exists(os.path.join(self.src_dir, self.bag_name)):
```

Comment:
@helrond I've introduced a bug here--if the bag is not actually in the directory, but is in the database, this is marking as successful. I'm not sure what we want the behavior to be, though:

1. Throw an error
2. Set the status back to the start, like how we've just updated Fornax

I'm inclined to go with 1 but didn't know if there was additional context you had?

Reply:
Either way, throwing an exception seems like a good starting place. If we're getting tons of errors from this, then you can take the approach we took in Fornax. But I think I agree with you that we'd want these failures to be a bit more noisy than waiting for transfers to finish processing...